virtual-desktop/src/app/authentication-manager/login/login.component.ts

```diff
@@ -183,14 +207,41 @@ export class LoginComponent implements OnInit {
     }    
   }
 
+  attemptPasswordReset(): void {
+    if (this.newPassword != this.confirmNewPassword) {
+      this.errorMessage = "New passwords do not match. Please try again.";
+    }  else {
+      this.authenticationService.performPasswordReset(this.username, this.password, this.newPassword, ZSS_AUTH).subscribe(
```

Comment:
You need to find out the plugin that can do the PW reset by reading the response to login. If you see "canChangePw" on any, record which one, and then use it here. If you find 0, you need to hide the pw panels, and if password=expired, then you just need to fail, rather than provide the reset option if it doesnt exist.